File: src/main/java/de/skit/grocy/recipe/dto/Recipe.java
package de.skit.grocy.recipe.dto;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

import de.skit.grocy.recipe.ingredient.dto.RecipeIngredient;
import de.skit.grocy.recipe.step.dto.RecipeStep;

public record Recipe(
    UUID id,
    UUID householdId,
    UUID createdBy,
    String title,
    String description,
    int baseServings,
    int version,
    Integer prepTimeMinutes,
    List<RecipeIngredient> ingredients,
    List<RecipeStep> steps,
    OffsetDateTime createdAt,
    OffsetDateTime updatedAt) {
}

File: src/vueapp/grocy-frontend/src/models/Recipe.ts
export type RecipeIngredient = {
	id?: string;
	name: string;
	quantity: number | null;
	unitText: string | null;
	sortIndex: number;
};

export type RecipeStep = {
	id?: string;
	instruction: string;
	sortIndex: number;
};

export type Recipe = {
	id: string;
	householdId: string;
	title: string;
	description?: string;
	baseServings: number;
	prepTimeMinutes?: number;
	ingredients: RecipeIngredient[];
	steps: RecipeStep[];
};

export type RecipeCreate = {
	title: string;
	description?: string;
	baseServings: number;
	prepTimeMinutes?: number;
	ingredients: RecipeIngredient[];
	steps: RecipeStep[];
};
